File: graphql-dgs-codegen-core/src/integTest/java/com/netflix/graphql/dgs/codegen/java/testcases/dataclasses/dataClassFieldDocs/expected/types/Movie.java
package com.netflix.graphql.dgs.codegen.java.testcases.dataclasses.dataClassFieldDocs.expected.types;

import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.Objects;

public class Movie {
  /**
   * The original, non localized title with some specials characters : %!({[*$,.:;.
   */
  private String title;

  public Movie() {
  }

  public Movie(String title) {
    this.title = title;
  }

  /**
   * The original, non localized title with some specials characters : %!({[*$,.:;.
   */
  public String getTitle() {
    return title;
  }

  public void setTitle(String title) {
    this.title = title;
  }

  @Override
  public String toString() {
    return "Movie{title='" + title + "'}";
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    Movie that = (Movie) o;
    return Objects.equals(title, that.title);
  }

  @Override
  public int hashCode() {
    return Objects.hash(title);
  }

  public static Builder newBuilder() {
    return new Builder();
  }

  public static class Builder {
    /**
     * The original, non localized title with some specials characters : %!({[*$,.:;.
     */
    private String title;

    public Movie build() {
      Movie result = new Movie();
      result.title = this.title;
      return result;
    }

    /**
     * The original, non localized title with some specials characters : %!({[*$,.:;.
     */
    public Builder title(String title) {
      this.title = title;
      return this;
    }
  }
}

File: graphql-dgs-codegen-core/src/test/resources/kotlin2/dataClassFieldDocs/expected/types/Movie.kt
package kotlin2.dataClassFieldDocs.expected.types

import com.fasterxml.jackson.`annotation`.JsonIgnoreProperties
import com.fasterxml.jackson.`annotation`.JsonProperty
import com.fasterxml.jackson.`annotation`.JsonTypeInfo
import com.fasterxml.jackson.databind.`annotation`.JsonDeserialize
import com.fasterxml.jackson.databind.`annotation`.JsonPOJOBuilder
import java.lang.IllegalStateException
import kotlin.String

@JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
@JsonDeserialize(builder = Movie.Builder::class)
public class Movie(
  title: () -> String? = titleDefault,
) {
  private val _title: () -> String? = title

  /**
   * The original, non localized title with some specials characters : %!({[*$,.:;.
   */
  public val title: String?
    get() = _title.invoke()

  public companion object {
    private val titleDefault: () -> String? = 
        { throw IllegalStateException("Field `title` was not requested") }

  }

  @JsonPOJOBuilder
  @JsonIgnoreProperties("__typename")
  public class Builder {
    private var title: () -> String? = titleDefault

    @JsonProperty("title")
    public fun withTitle(title: String?): Builder = this.apply {
      this.title = { title }
    }

    public fun build() = Movie(
      title = title,
    )
  }
}
